### ISSUE TITLE
A11y - tab navigation out of order for webviews

### ISSUE THREAD

**Issue by alanrenmsft** [microsoft/vscode#166768]
Repro Steps:

1. follow the instructions in https://github.com/microsoft/vscode-webview-ui-toolkit-samples/tree/main/default/hello-world
2. run the hello world command to launch the webview
3. press tab key and observe the order of the controls getting the focus

expected:
![image](https://user-images.githubusercontent.com/13777222/202823225-838c3b21-00b2-4739-abc5-8215da376784.png)

actual:
![image](https://user-images.githubusercontent.com/13777222/202823128-972b77c1-8931-4072-8f66-2f31d766f0bb.png)

![tab-order](https://user-images.githubusercontent.com/13777222/202822786-75a3e572-d636-42a8-b429-692f9de5539f.gif)

**Comment by fsteffi** [comment]
**GitHub Tags:**#A11yMAS;#A11yTCS;#Visual Studio Code Client;#BM-VisualStudioCodeClient-Win32-Aug2022;#Win32;#DesktopApp;#A11ySev2;#Keyboard;#AINotInScope;#FocusOrder;#WCAG2.4.3;#Win11;

**Comment by meganrogge** [comment]
@mjbvz does this look like something that can be fixed easily? A code pointer would be good if you want me to look into it

**Comment by mjbvz** [comment]
@meganrogge I fixed one possible cause of this #168541 but it doesn't fix this problem in my testing

Not sure if this ever worked TBH. We're trying to use `aria-flowto` to fix tab order but it's unclear how well supported this is. If that doesn't work, not sure what else we can do since we can't change the dom structure

**Comment by isidorn** [comment]
@mjbvz thanks for looking into this. 
Did you have a chance to tryout `aria-flowto` attribute to fix tab order?
If that does not work we should have more clarity on the limitations here, and if this is something we should file against the aria specification maybe?
An alternative is to have guidance and recommendation on how to solve this for extension authors that contribute WebViews.

**Comment by alanrenmsft** [comment]
@mjbvz any update?

**Comment by mjbvz** [comment]
It's unclear how we can fix this. We are using `aria-flowto` and cannot change the dom structure. It the tab order is still wrong, I'd say this could be a browser bug but none of the browsers support it and I'm familiar enough with the spec to say for sure this is an issue

Here's a minimal example:

```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <button>1st</button>

    <div aria-flowto="section">flow to div</div>

    <button>last</button>

    <section id="section">
        <button>flow to button</button>
    </section>
</body>
</html>
```

Will tentatively open an issue against chromium about this

**Comment by mjbvz** [comment]
Looks like `aria-flowto` has very poor support overall: https://a11ysupport.io/tech/aria/aria-flowto_attribute

**Comment by mjbvz** [comment]
Chromium bug: https://bugs.chromium.org/p/chromium/issues/detail?id=1413786

**Comment by scottaohara** [comment]
aria-flowto wouldn't actually do anything to fix the focus order though.  aria-flowto would be for how someone using the virtual cursor (or scan mode) should be informed to navigate to different elements, but modifying this would do absolutely nothing to change the focus order.

**Comment by isidorn** [comment]
@scottaohara thanks for your comment. Do you have suggestions on how to fix?

**Comment by scottaohara** [comment]
i'm not exactly sure without understanding this a bit more.  re: the comment that the DOM structure cannot be altered has be a bit concerned about the practicality of any potential path forward.  might be a good thing to chat about.

**Comment by isidorn** [comment]
Thanks. Let's setup a time to talk. I will reach out to you and Matt.

**Comment by isidorn** [comment]
@mjbvz and me have synced with @scottaohara and after discussion we have agreed that there is a valid way to navigate through the workbench (including WebViews) in the expected way by using the `workbench.action.focusNextPart` and `workbench.action.focusPreviousPart` commands.
We have agreed that this limitation needs to be documented and we have just done that here https://github.com/microsoft/vscode-docs/blob/vnext/docs/editor/accessibility.md#tab-navigation

Let me know if you have any feedback on the docs. I expect them to be live on our website later this week.

We have also agreed since there are valid workaround for this limitation that this is indeed a feature request and can be marked as Sev3. Thus I am updating the labels of this issue.

Thanks all!

**Comment by VSCodeTriageBot** [comment]
We closed this issue because we [don't plan to address it](https://aka.ms/vscode-out-of-scope) in the foreseeable future. If you disagree and feel that this issue is crucial: we are happy to listen and to reconsider.

If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Thanks for your understanding, and happy coding!
